syntax = "proto3";
package notification;

message NotifyObjectResponse {
  int64 id = 1;
  string client_id = 2;
  string client_workspace_domain = 3;
  string ref_client_id = 4;
  string ref_workspace_domain = 5;
  string ref_subject_name = 6;
  int64 ref_group_id = 7;
  string notify_type = 8;
  string notify_image = 9;
  string notify_title = 10;
  string notify_content = 11;
  bool read_flg = 12;
  int64 created_at = 13;
}


message BaseResponse {
    string error = 1;
}

// Request: get list notify
message Empty {
}
// Response: GetNotifiesResponse
message GetNotifiesResponse {
  repeated NotifyObjectResponse lst_notify = 1;
}


// Request: subcribe
message SubscribeRequest {
  string device_id = 1;
}
//response: BaseResponse

// Request: un-subcribe
message UnSubscribeRequest {
  string device_id = 1;
}
//response: BaseResponse

// Request: listen
message ListenRequest {
  string device_id = 1;
}
//response: NotifyObjectResponse stream

message ReadNotifyRequest {
  string notify_id = 1;
}
//response: BaseResponse


// Method
service Notify {
    rpc read_notify (ReadNotifyRequest) returns (BaseResponse);
    rpc get_unread_notifies (Empty) returns (GetNotifiesResponse);
    rpc subscribe (SubscribeRequest) returns (BaseResponse);
    rpc un_subscribe (UnSubscribeRequest) returns (BaseResponse);
    rpc listen (ListenRequest) returns (stream NotifyObjectResponse);
}
